plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -116,6 +116,12 @@ data DefaultFun
     | MkPairData
     | MkNilData
     | MkNilPairData
+    -- Show
+    | ShowBool
+    | ShowInteger
+    | ShowString
+    | ShowByteString
+    | ShowData
```

Comment:
Let's think about what it would mean to try and cost a single builtin like that. It is, ultimately, doing typecase, because it will actually be delegating to a different implementation for every builtin type. So the behavour is massively heterogeneous. Well, as you said, maybe it's always linear in the size? That would be nice, but it would have to a) be true for all our builtin types, and b) be true for _all builtin types we ever add in the future_, i.e. we're binding our future selves in a potentially very annoying way. Seems sketchy to me.

Reply:
> Actually, it probably doesn't really complicate anything once the thing that I'm currently working on gets merged.

I.e. we can keep the five distinct costing function, benchmark them separately etc and then merge them into one costing function at the definition site of the builtin. But I haven't checked this yet.

Reply:
I'm not favoring one approach over another; just trying to cost these show functions separately, so that we can see if their cost models are similar enough to justify combining them into a single function.

If we start with costing a single function, and find out the cost is non-linear or have some weird shape, will it be easy to figure out which part went wrong? I don't know, @kwxm should know better.

Reply:
> I.e. we can keep the five distinct costing function, benchmark them separately etc and then merge them into one costing function at the definition site of the builtin. But I haven't checked this yet.

That seems like it makes our costing infrastructure *much* more complicated....